app/src/main/java/org/lineageos/twelve/repositories/ResumptionPlaylistRepository.kt

```diff
@@ -0,0 +1,52 @@
+package org.lineageos.twelve.repositories
+
+import org.lineageos.twelve.database.TwelveDatabase
+import org.lineageos.twelve.models.ResumptionPlaylist
+
+/**
+ * Manages the playlist used when the user wants to resume playback from the last queue.
+ */
+class ResumptionPlaylistRepository(val database: TwelveDatabase) {
+    /**
+     * Get the last resumption playlist or an empty one.
+     */
+    suspend fun getResumptionPlaylist() =
+        database.getResumptionPlaylistDao().getResumptionPlaylistWithItems()?.let {
+            val sortedItems = it.items.sortedBy { item -> item.playlistIndex }
+            // This is needed to handle holes between indexes
```

Comment:
Is this even possible? you insert them via forEachIndexed so indices _are_ without holes per definition no?
Also, you can fix the order directly in sqlite by adding `ORDER BY id`